src/EditorFeatures/CSharpTest/InlineMethod/CSharpInlineMethodTests.cs

```diff
@@ -0,0 +1,1892 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InlineMethod;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.InlineMethod
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsInlineMethod)]
+    public class CSharpInlineMethodTests
+    {
+        private class TestVerifier : CSharpCodeRefactoringVerifier<CSharpInlineMethodRefactoringProvider>.Test
+        {
+            public static async Task TestInRegularAndScript1Async(
+                string initialMarkUp,
+                string expectedMarkUp,
+                int index = 0)
+            {
+                var test = new TestVerifier { CodeActionIndex = index };
+                test.TestState.Sources.Add(initialMarkUp);
+                test.FixedState.Sources.Add(expectedMarkUp);
+                await test.RunAsync().ConfigureAwait(false);
+            }
+        }
+
+        [Fact]
+        public Task TestInlineMethodWithSingleStatement()
+            => TestVerifier.TestInRegularAndScript1Async(
+                @"
+public class TestClass
+{
+    private void Caller(int i, int j)
+    {
+        Ca[||]llee(i, j);
```

Comment:
in terms of the feature, i see we have `inline method`.  do we have `inline, then delete method` and `inline method into all callers`?

Reply:
the reason i ask is that i see these examples and the caller is updated fine.  but i can see users saying "ok.. but now i have to go delete teh callee, i want that to optionally happen as well".

Reply:
I think that could be provided maybe via the nested code action.